### PR TITLE
HMAN 598: Second update to a previously listed hearing gives a 500 error

### PR DESCRIPTION
**Pull request by AntonyLeons**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/HMAN-598

### Change description ###

src/main/java/uk/gov/hmcts/reform/hmc/data/HearingResponseEntity.java
has the change required to prevent the null error.

other changes are bumping cftlib and style fixes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/data/HearingResponseEntity.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ public class HearingResponseEntity extends BaseEntity implements Serializable {
     private String listingTransactionId;
 
     public Optional<HearingDayDetailsEntity> getEarliestHearingDayDetails() {
-        return getHearingDayDetails().stream()
+        return getHearingDayDetails().stream().filter(h -> h.getStartDateTime() != null)
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/reform/hmc/data/HearingResponseEntity.java:93]: can you add test scenario with start date time null

File: src/main/java/uk/gov/hmcts/reform/hmc/validator/LinkedHearingValidator.java [hunk 2]
```diff
@@ -366,8 +366,8 @@ public LocalDate filterHearingResponses(HearingEntity hearingEntity) {
         if (log.isDebugEnabled()) {
             if (hearingResponse.isPresent()) {
                 log.debug("hearing response: {} : {}",
-                        hearingResponse.get().getHearingResponseId(),
-                        hearingResponse.get().getRequestTimeStamp());
+                          hearingResponse.get().getHearingResponseId(),
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/reform/hmc/validator/LinkedHearingValidator.java:369]: No changes in the code. please remove the extra spaces.

File: src/test/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedControllerTest.java [hunk 3]
```diff
@@ -47,7 +47,7 @@
 @ExtendWith(SpringExtension.class)
 @WebMvcTest(controllers = PartiesNotifiedController.class,
         excludeFilters = @ComponentScan.Filter(type = ASSIGNABLE_TYPE, classes =
-                {SecurityConfiguration.class, JwtGrantedAuthoritiesConverter.class}))
+            {SecurityConfiguration.class, JwtGrantedAuthoritiesConverter.class}))
```
Comment (padmapriyanalam) [inline src/test/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedControllerTest.java:50]: No changes in the code. please remove the extra spaces.

File: src/test/java/uk/gov/hmcts/reform/hmc/validator/CapitalizedEnumPatternValidatorTest.java [hunk 4]
```diff
@@ -28,7 +28,7 @@ void setUp() {
 
     @ParameterizedTest
     @ValueSource(strings = {"MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY", "MoNdAy",
-            "TUEsday", "wednesday", "THuRSDAY", "FridaY", "saturday", "sUNDAY"})
+        "TUEsday", "wednesday", "THuRSDAY", "FridaY", "saturday", "sUNDAY"})
```
Comment (padmapriyanalam) [inline src/test/java/uk/gov/hmcts/reform/hmc/validator/CapitalizedEnumPatternValidatorTest.java:31]: No changes in the code. please remove the extra spaces.

File: src/main/java/uk/gov/hmcts/reform/hmc/validator/LinkedHearingValidator.java (hunk 2 above)
Comment (AntonyLeons) [reply src/main/java/uk/gov/hmcts/reform/hmc/validator/LinkedHearingValidator.java:369]: so strange behaviour when upgrading cftlib, these are required or style checks fail
